frontend/public/components/secrets/create-secret/utils.ts

```diff
@@ -94,3 +97,48 @@ export const useSecretDescription = (typeAbstraction: SecretTypeAbstraction): st
       return null;
   }
 };
+
+export const opaqueEntriesToObject = (opaqueEntriesArray: OpaqueDataEntry[]): SecretChangeData => {
+  return (opaqueEntriesArray ?? []).reduce(
+    (acc, { key, value, isBinary_ }) => {
+      return {
+        stringData: {
+          ...acc.stringData,
+          [key]: isBinary_ ? null : value,
+        },
+        base64StringData: {
+          ...acc.base64StringData,
+          [key]: !isBinary_ ? Base64.encode(value) : value,
```

Comment:
Do we need to encode and store non-binary values here? They are already being stored in stringData, right?
```suggestion
          [key]: !isBinary_ ? null : value,
```

Reply:
Current implementation counted on base64StringData being always populated, as `opaqueSecretObjectToArray` function is used during the state initialization of the `OpaqueSecretForm`'s state ( data entries ). As part of this function we determine if the contents of the entry are binary or not. To check this I always checked the contents stored in the  `base64StringData` as so:
`const isValueBinary = isBinary(null, Buffer.from(base64StringData[key], 'base64'));`

We could go in the direction of your suggestion, but we would have to omit the whole `[key]: value` pair in case the data is non-binary, like so:
```suggestion
          base64StringData: isBinary_
          ? { ...acc.base64StringData, [key]: value }
          : acc.base64StringData,
```
This needs to be done as the null value for the given key in base64StringData would [overwrite](https://github.com/openshift/console/blob/4900dadc97aaee9e4c0526fe21d815796746c7d7/frontend/public/components/secrets/create-secret/SecretFormWrapper.tsx#L81) the one stored in the `stringData` during the save action ( as the same property from the second spread has priority before the first spread ).

And then one small change would be needed `opaqueSecretObjectToArray`, see the suggestions down bellow.

Reply:
Hmm, maybe we only use base64 endcoded data for this form?

Reply:
That is definitely doable, I tested it out locally and got it to work and It would result in less code, but I'm not sure if it's best practice to encode string data the users type into the input fields.

Reply:
I think it's better than tracking input state in two different formats. That approach seems more bug-prone to me.

Reply:
Agreed! Will update the PR.